src/lib.rs

```diff
@@ -2,23 +2,36 @@
 
 pub mod chainpack;
 pub mod cpon;
+pub mod json;
 pub mod datetime;
 pub mod decimal;
 pub mod metamap;
 pub mod reader;
 pub mod rpcvalue;
 pub mod util;
 pub mod writer;
+mod textrdwr;
 
 pub use datetime::DateTime;
 pub use decimal::Decimal;
 pub use metamap::MetaMap;
 pub use reader::{ReadError, ReadResult, Reader};
 pub use rpcvalue::Value;
-pub use rpcvalue::{Blob, List, Map, RpcValue};
+pub use rpcvalue::{Blob, List, Map, IMap, RpcValue};
 pub use writer::{WriteResult, Writer};
 
 pub use chainpack::{ChainPackReader, ChainPackWriter};
 pub use cpon::{CponReader, CponWriter};
+pub use json::{JsonReader, JsonWriter};
 
 pub use libshvproto_macros::TryFromRpcValue;
+
+fn to_hex(b: u8) -> u8 {
```

Comment:
This function actually converts a nibble to its hex representation, but it is not clear from the name nor the parameter of the function. Furthermore, there is no input validation which makes the function error prone.

I suggest this change:
```rust
fn u8_to_hex(num: u8) -> (u8, u8) {
    let high = num / 16;
    let low = num % 16;
    (high, low)
}
```